stubs/WTForms/wtforms/fields/core.pyi

```diff
@@ -0,0 +1,133 @@
+    # by type checkers, so we use a descriptor hack to get around this
+    # limitation instead
+    def __new__(cls, *args: Any, **kwargs: Any) -> Self: ...
+    def __init__(
+        self,
+        label: str | None = None,
+        # for tuple we can be a bit more type safe and only accept validators
+        # that would work on this or a less specific field, but in general it
+        # would be too annoying to restrict to Sequence[_Validator], since mypy
+        # will infer a list of mixed validators as list[object], since that is
+        # the common base class between all validators
+        validators: tuple[_Validator[_FormT, Self], ...] | list[Any] | None = None,
+        filters: Sequence[_Filter] = (),
+        description: str = "",
+        id: str | None = None,
+        default: object | None = None,
+        widget: _Widget[Self] | None = None,
+        render_kw: dict[str, Any] | None = None,
+        name: str | None = None,
+        _form: BaseForm | None = None,
+        _prefix: str = "",
+        _translations: _SupportsGettextAndNgettext | None = None,
+        _meta: DefaultMeta | None = None,
+    ) -> None: ...
+    def __html__(self) -> str: ...
+    def __call__(self, **kwargs: object) -> Markup: ...
+    @classmethod
+    def check_validators(cls, validators: Iterable[_Validator[_FormT, Self]] | None) -> None: ...
+    def gettext(self, string: str) -> str: ...
+    def ngettext(self, singular: str, plural: str, n: int) -> str: ...
+    def validate(self, form: BaseForm, extra_validators: tuple[_Validator[_FormT, Self], ...] | list[Any] = ()) -> bool: ...
+    def pre_validate(self, form: BaseForm) -> None: ...
+    def post_validate(self, form: BaseForm, validation_stopped: bool) -> None: ...
+    def process(
+        self, formdata: _MultiDictLikeWithGetlist | None, data: Any = ..., extra_filters: Sequence[_Filter] | None = None
+    ) -> None: ...
+    def process_data(self, value: Any) -> None: ...
+    def process_formdata(self, valuelist: list[Any]) -> None: ...
+    def populate_obj(self, obj: object, name: str) -> None: ...
+
+    # this is a workaround for what is essentialy illegal in static type checking
+    # Field.__new__ would return an UnboundField, unless the _form parameter is
+    # specified. We can't really work around it by making UnboundField a subclass
+    # of Field, since all subclasses of Field still need to return an UnboundField
+    # and we can't expect third parties to add a __new__ method to every field
+    # they define...
+    # This workaround only works for Form, not BaseForm, but we take what we can get
+    # BaseForm shouldn't really be used anyways
+    @overload
+    def __get__(self, obj: None, owner: type[object] | None = None) -> UnboundField[Self]: ...
+    @overload
+    def __get__(self, obj: object, owner: type[object] | None = None) -> Self: ...
+
+    # we put this at the bottom so it doesn't shadow builtins.type
```

Comment:
I don't think that makes sense in stubs

Reply:
`type` will be shadowed for the rest of the class body if i move it above so I either need to use `typing.Type` or define my own alias for the rest of the annotations in the class body that use `builtins.type`, so this seemed more pragmatic.

Reply:
Mypy and pyright have different behaviour here (and pyright's behaviour arguably makes more sense, actually), so it's best not to make any assumptions on how this kind of thing will be interpreted by the type checker. See https://github.com/microsoft/pyright/issues/4856

Reply:
That's fair. I will change it accordingly.